src/account_manager/operations/stronghold_backup/mod.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fs, path::PathBuf, sync::atomic::Ordering};
+
+use iota_client::secret::{stronghold::StrongholdSecretManager, SecretManager, SecretManagerDto};
+use zeroize::Zeroize;
+
+use crate::account_manager::{AccountHandle, AccountManager, AccountManagerBuilder};
+
+mod stronghold_snapshot;
+use stronghold_snapshot::{read_data_from_stronghold_snapshot, store_data_to_stronghold};
+
+impl AccountManager {
+    /// Backup the account manager data in a Stronghold file
+    /// stronghold_password must be the current one when Stronghold is used as SecretManager.
+    pub async fn backup(&self, backup_path: PathBuf, mut stronghold_password: String) -> crate::Result<()> {
+        log::debug!("[backup] creating a stronghold backup");
+        let mut secret_manager = self.secret_manager.write().await;
+
+        let secret_manager_dto = SecretManagerDto::from(&*secret_manager);
+
+        match &mut *secret_manager {
+            // Backup with existing stronghold
+            SecretManager::Stronghold(stronghold) => {
+                stronghold.set_password(&stronghold_password).await?;
+
+                store_data_to_stronghold(self, stronghold, secret_manager_dto).await?;
+
+                // Get current snapshot_path to set it again after the backup
+                let current_snapshot_path = stronghold.snapshot_path.clone();
+
+                // Set the snapshot path
+                stronghold.snapshot_path = backup_path;
+
+                // Write snapshot to backup path
+                stronghold.write_stronghold_snapshot().await?;
+
+                // Reset snapshot_path
+                stronghold.snapshot_path = current_snapshot_path;
+            }
+            // Backup with new stronghold
+            _ => {
+                let mut backup_stronghold = StrongholdSecretManager::builder()
+                    .password(&stronghold_password)
+                    .try_build(backup_path)?;
+
+                store_data_to_stronghold(
+                    self,
+                    // If the SecretManager is not Stronghold we'll create a new one for the backup
```

Comment:
Is this really where we want to put this comment ? Feels a bit weird, maybe before creating it ?

Reply:
Right, moved the generation, but didn't move the comment